gui/gui/src/ts/components/pages/NodeSearchPage.tsx

```diff
@@ -1,187 +1,97 @@
-  };
-
-  const handleBackButtonClick = (
-    event: React.MouseEvent<HTMLButtonElement>
-  ) => {
-    onChangePage(event, page - 1);
-  };
-
-  const handleNextButtonClick = (
-    event: React.MouseEvent<HTMLButtonElement>
-  ) => {
-    onChangePage(event, page + 1);
-  };
-
-  const handleLastPageButtonClick = (
-    event: React.MouseEvent<HTMLButtonElement>
-  ) => {
-    onChangePage(event, Math.max(0, Math.ceil(count / rowsPerPage) - 1));
-  };
+export const NodeSearchPage: React.FunctionComponent<{}> = ({}) => {
+  const [state, setState] = React.useState<MatchingNodesQueryState>({
+    matchingNodes: [],
+    searchText: "",
+    count: 0,
+    offset: 0,
+    queryLimit: 10,
+  });
 
-  return (
-    <div className={classes.root}>
-      <IconButton
-        onClick={handleFirstPageButtonClick}
-        disabled={page === 0}
-        aria-label="first page"
-      >
-        {theme.direction === "rtl" ? (
-          <FontAwesomeIcon icon={faAngleDoubleRight} />
-        ) : (
-          <FontAwesomeIcon icon={faAngleDoubleLeft} />
-        )}
-      </IconButton>
-      <IconButton
-        onClick={handleBackButtonClick}
-        disabled={page === 0}
-        aria-label="previous page"
-      >
-        {theme.direction === "rtl" ? (
-          <FontAwesomeIcon icon={faAngleRight} />
-        ) : (
-          <FontAwesomeIcon icon={faAngleLeft} />
-        )}
-      </IconButton>
-      <IconButton
-        onClick={handleNextButtonClick}
-        disabled={page >= Math.ceil(count / rowsPerPage) - 1}
-        aria-label="next page"
-      >
-        {theme.direction === "rtl" ? (
-          <FontAwesomeIcon icon={faAngleLeft} />
-        ) : (
-          <FontAwesomeIcon icon={faAngleRight} />
-        )}
-      </IconButton>
-      <IconButton
-        onClick={handleLastPageButtonClick}
-        disabled={page >= Math.ceil(count / rowsPerPage) - 1}
-        aria-label="last page"
-      >
-        {theme.direction === "rtl" ? (
-          <FontAwesomeIcon icon={faAngleDoubleLeft} />
-        ) : (
-          <FontAwesomeIcon icon={faAngleDoubleRight} />
-        )}
-      </IconButton>
-    </div>
-  );
-}
+  const onLoadedData = (updatedState: Partial<MatchingNodesQueryState>) =>
+    setState((prevState) => ({...prevState, ...updatedState}));
 
-export const NodeSearchPage: React.FunctionComponent<{}> = ({}) => {
-  const searchTextInputRef = React.useRef<HTMLInputElement>(null);
+  const apolloClient = useApolloClient();
 
-  const [searchText, setSearchText] = React.useState<string>("");
-  const [matchingNodesPage, setMatchingNodesPage] = React.useState<number>(0);
-  const [matchingNodesQueryLimit, setMatchingNodesQueryLimit] = React.useState<
-    number
-  >(10);
+  const onSearchSubmit = (searchText: string) =>
+    apolloClient
+      .query<MatchingNodesQuery, MatchingNodesQueryVariables>({
+        query: MatchingNodesQueryDocument,
+        variables: {
+          limit: state.queryLimit,
```

Comment:
Not entirely sure why you're keeping state here, since this is all going to migrate to the query string. But save it for the next PR.